core/app/models/spree/payment/cancellation.rb

```diff
@@ -0,0 +1,60 @@
+module Spree
+  class Payment
+    # Payment cancellation handler
+    #
+    # Cancels a payment by trying to void first and if that fails
+    # creating a refund about the full amount instead.
+    #
+    class Cancellation
+      DEFAULT_REASON = 'Order canceled'.freeze
+
+      attr_reader :reason
+
+      # @param reason [String] (DEFAULT_REASON) -
+      #   The reason used to create the Spree::RefundReason
+      def initialize(reason: DEFAULT_REASON)
```

Comment:
Maybe this should not be an optional argument since we are passing it on [initialization](https://github.com/solidusio/solidus/pull/2111/files#diff-a9fae1594c22d1ec005ac8501d80d7afR362).